packages/terra-consumer-nav/src/components/nav-items/NavItem.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import styles from './NavItem.scss';
+import NavToggler from '../nav-toggler/NavToggler';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  slug: PropTypes.string,
+  navType: PropTypes.oneOf(['grouping', 'external', 'application', 'modal']),
+  target: PropTypes.string,
+  uri: PropTypes.string.isRequired,
+  text: PropTypes.string.isRequired,
+  isActive: PropTypes.bool,
+  badgeValue: PropTypes.number,
+
+  toggleId: PropTypes.number,
+  handleToggle: PropTypes.func,
+  isOpen: PropTypes.bool,
+  /*
+    TODO: Need to figure out what we are doing with icons or if navitems should even have icons
+
+    icon: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.element,
+    ]),
+  */
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  isActive: false,
+  handleToggle: () => {},
+};
+
+const NavItem = ({
+  slug,
+  navType,
+  target,
+  uri,
+  text,
+  isActive,
+  badgeValue,
+  toggleId,
+  handleToggle,
+  isOpen,
+  children,
+  ...customProps
+}) => {
+  const activeClass = cx([
+    { active: isActive },
+  ]);
```

Comment:
actually class names don't need to be within an array. it will be flattened by the classnames method if provided
https://www.npmjs.com/package/classnames#usage